### DISCUSSION TITLE
How to record the PyBullet Gym GUI?

### DISCUSSION THREAD

**Discussion by btjanaka** [bulletphysics/bullet3#4019] [Q&A] [answered]
Hi, I am trying to record videos of PyBullet Gym environments. So far, I have been able to save frames with `render(mode="rgb_array")` and then combine these frames with `ffmpeg`, but this method is pretty slow, and the output does not look as "pretty" as the one in the PyBullet GUI. Thus, I have also tried to record the GUI with the `Monitor` wrapper from OpenAI Gym, but this only produces an empty MP4 file. Finally, I have tried using screenshot software (Kazam), but the frame rate is pretty slow in this method. Any other suggestions on how to record a video of the PyBullet GUI?

**Comment by btjanaka** [comment]
I just tried screen capture again with Kazam, and the frame rate seems alright. Still open to other suggestions though.

**Comment by btjanaka** [comment, marked as answer]
Figured out a solution that uses pyvirtualdisplay -- essentially, you can render to a SmartDisplay (https://github.com/ponty/pyvirtualdisplay/tree/2.2#screenshot) and then take screenshots after every frame. Finally, you can assemble the frames with ffmpeg.

**Comment by rohit-kumar-j** [comment]
@btjanaka, Pybullet Quickstart Guide: `pybullet.startStateLogging`
>For MP4 video recording you can use the logging option STATE_LOGGING_VIDEO_MP4. We
plan to implement various other types of logging, including logging the state of VR controllers
